### DISCUSSION TITLE
Cannot chmod +x files under /mnt/c/Windows

### DISCUSSION THREAD

**Discussion by dac514** [microsoft/WSL#8464] [General]
### Version

Microsoft Windows [Version 10.0.22000.708]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.10.16.3-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Apr 2 22:23:49 UTC 2021

### Distro Version

Ubuntu 22.04

### Other Software

N/A

### Repro Steps

My `/etc/wsl.conf` is:

```
[automount]
options = "metadata,umask=022,fmask=111,case=off"
```

When I run [wslsys](https://wslutiliti.es/wslu/man/wslsys.html) I get: 

```
/usr/bin/wslsys: line 216: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 156: /mnt/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 279: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslsys: line 265: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslsys: line 216: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslsys: line 274: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslvar: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslvar: line 156: /mnt/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe: Permission denied
/usr/bin/wslvar: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 156: /mnt/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 284: /mnt/c/Windows/System32/reg.exe: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
/usr/bin/wslsys: line 156: /mnt/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe: Permission denied
/usr/bin/wslsys: line 149: /mnt/c/Windows/System32/chcp.com: Permission denied
WSL Version: 1
Locale:
Release Install Date: Wed Dec 31 19:00:00 EST 1969
Branch:
Build:
Full Build:
Display Scaling:
Windows Theme: dark
Windows Uptime: 0d 0h 0m
WSL Uptime: 0d 0h 32m
WSL Release: Ubuntu 22.04 LTS
WSL Kernel: Linux 5.10.16.3-microsoft-standard-WSL2
Packages Count: 695
```

No problem, I think to myself, I'll just **_run as administrator_** and do:

```
 chmod +x  /mnt/c/Windows/System32/reg.exe
```

### Expected Behavior

I can change the exec permission.

### Actual Behavior

`chmod: changing permissions of '/mnt/c/Windows/System32/reg.exe': Permission denied`

**Comment by elsaco** [comment]
@connerbw not sure why you'd want to change permissions in `C:\Windows\System32` folder, but WSL already mounts `C:\` with execute enabled:

```
tux@RIPPER ~ $ getfacl /mnt/c/Windows/System32/reg.exe
getfacl: Removing leading '/' from absolute path names
# file: mnt/c/Windows/System32/reg.exe
# owner: tux
# group: tux
user::r-x
group::r-x
other::r-x

tux@RIPPER ~ $ ls -l /mnt/c/Windows/System32/reg.exe
-r-xr-xr-x 2 tux tux 77312 Dec  7  2019 /mnt/c/Windows/System32/reg.exe
```

**Comment by dac514** [comment]
@elsaco the important part is under the **_Repro steps_** please re-read?

```
 getfacl /mnt/c/Windows/System32/reg.exe
getfacl: Removing leading '/' from absolute path names
# file: mnt/c/Windows/System32/reg.exe
# owner: dac
# group: dac
user::r--
group::r--
other::r--
```

**Comment by dac514** [comment]
Why:

- https://gist.github.com/carlolars/7914d2606204001e798c812961c3ac88
- https://kizu514.com/blog/ubuntu-22-04-lts-inside-windows-11/

Similar issues:

- https://github.com/microsoft/WSL/issues/3165
- https://github.com/microsoft/WSL/issues/4567

Not sure why this was converted into a "_discussion_" ... 👎

**Reply by sebble** [reply]
Remounting the rest of `c/Windows` has helped me, this restores the unwanted but necessary file permissions within the unmodifiable c/Windows directory.

```
sudo mount -t drvfs c:/Windows /mnt/c/Windows
```

**Reply by jmelanso** [reply]
This fixed the issue for me!

**Comment by NotTheDr01ds** [comment]
@sebble Good workaround.  

@connerbw

> Not sure why this was converted into a "discussion" ... 👎

I'm assuming that it was converted into a discussion because it isn't a bug -- It's expected behavior.

WSL runs at the permission level of the *Windows* user.  As such, changing files in `C:\Windows` (and some other directories) (even the NTFS extended attributes in which WSL stores the Linux permissions) is off-limit to WSL (even `sudo`/`root`) when running under your normal account.   The behavior is noted in the gist you linked to.  While it recommends running WSL as Administrator, I would caution against that.

Much better and safer to use @sebble's workaround.
